src/main/java/com/example/demo/controllers/AuctionController.java

```diff
@@ -0,0 +1,28 @@
+package com.example.demo.controllers;
+
+import com.example.demo.dtos.AuctionDTO;
+import com.example.demo.entities.User;
+import com.example.demo.services.AuctionService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequiredArgsConstructor
+public class AuctionController {
+    private final AuctionService auctionService;
+
+    @PostMapping("/auction")
+    public void createAuction(@AuthenticationPrincipal User owner,
```

Comment:
no request params should be passed for POST, auth is better to be handled elsewhere